sys/sys/proc.h

```diff
@@ -898,6 +898,11 @@ struct proc {
 #define	P2_MEMBAR_GLOBE		0x00400000	/* membar global expedited
 						   registered */
 
+#define	P2_CHERI_C18N_ENABLE	0x10000000	/* Force enable compartmentalisation */
```

Comment:
FWIW, I view the P2_* flags as part of the procctl() implementation, so would probably move the those bits to the next commit and only have this commit be adding the rtld bits and the new flag in elf_common.h.